heat/core/operations.py

```diff
@@ -544,7 +549,620 @@ def sum(x, axis=None, out=None):
 
     return __reduce_op(x, torch.sum, MPI.SUM, axis, out)
 
-  
+
+def merge_means(mu1, n1, mu2, n2):
+    """
+    Function to merge two means by pairwise update
+
+    Parameters
+    ----------
+    mu1 : 1D ht.tensor or 1D torch.tensor
+          Calculated mean
+    n1 : 1D ht.tensor or 1D torch.tensor
+         number of elements used to calculate mu1
+    mu2 : 1D ht.tensor or 1D torch.tensor
+          Calculated mean
+    n2 : 1D ht.tensor or 1D torch.tensor
+         number of elements used to calculate mu2
+
+    Returns
+    -------
+    mean of combined set
+    number of elements in the combined set
+    """
+    if mu1.item() == np.nan:
+        mu1 = 0.0
+    if mu2.item() == np.nan:
+        mu2 = 0.0
+    delta = mu2.item() - mu1.item()
+    n1 = n1.item()
+    n2 = n2.item()
+    return mu1 + n2 * (delta / (n1+n2)), n1 + n2
+
+
+def mean(x, dimen=None, all_procs=False):
+    """
+    Calculates and returns the mean of a tensor.
+    If a dimension is given, the mean will be taken in that direction.
+
+    Parameters
+    ----------
+    x : ht.tensor
+        Values for which the mean is calculated for
+    dimen : None, Int, iterable
+            Dimension which the mean is taken in.
+            Default: None -> mean of all data calculated
+    all_procs : Bool
+                Flag to distribute the data to all processes
+                If True: will split the result in the same direction as x
+                Default: False (mean of the whole dataset still calculated but not on every node)
+
+    Returns
+    -------
+    ht.tensor containing the mean/s, if split, then split in the same direction as x.
+    """
+
+    def combine_all_means(target_dimens):
```

Comment:
target_dimens should be axis

Reply:
what i meant by this is that it is the dimensions of the output tensor and not the axis given to the function

Reply:
Alright, my bad. Could we call out output_shape to be consistent with numpy speak? Also, it there a need for the inline function except the parameter closure?

Reply:
There isnt a true need for the inline functions but i didnt want to have to put the same code twice if it's not needed. 

output_shape change is made